mula/scheduler/schedulers/boefje.py

```diff
@@ -368,6 +404,40 @@ def is_task_allowed_to_run(self, boefje: Plugin, ooi: OOI) -> bool:
 
         return True
 
+    def is_task_rate_limited(self, task: BoefjeTask) -> bool:
+        """Checks whether a task is rate limited.
+
+        Args:
+            task: The task to check.
+
+        Returns:
+            True if the task is rate limited, False otherwise.
+        """
+        if task.boefje.rate_limit is None:  # TODO: this needs to be returned by a boefje manifest
+            return False
+
+        # Check if namespace is already in the tasks if not add it
+        if self.delayed_tasks.get(task.hash) is None:
```

Comment:
Isn't as straightforward as expect since a `Task` expects a `p_item`, which means it is evaluated and has been score. But when the task is allowed on the queue that ranking may not necessarily correspond anymore

Reply:
We do probably want to create a task for it, because then we're able to communicate to the user that the specific task is being delayed, otherwise in this implementation, this will be obfuscated